scripts/orchestra_methods.py

```diff
@@ -0,0 +1,319 @@
+    call_args = list()
+    call_args.append('cellranger')
+    call_args.append('mkfastq')
+    call_args.append('--run=' + os.path.join(run,""))
+    call_args.append('--csv=' + samples_csv)
+    call_args.append('--output-dir=fastqs')
+    call(call_args)
+
+    # move the fastqs to the output directory
+    call(["gsutil", "-q", "-m", "mv", os.path.join('fastqs',flowcell), output_directory])
+    # move and rename the qc summary file to fastq output directory, rename it so it doesn't get rewritten if there are multiple bcls
+    call(["gsutil", "-q", "-m", "mv", os.path.join(flowcell,"outs","qc_summary.json"), os.path.join(output_directory,flowcell+"_qc_summary.json")])
+    
+    # write the path of fastqs for cromwell
+    file = open("path.txt","w") 
+    file.write(os.path.join(output_directory,flowcell,""))
+    file.close()
+
+    # move the undetermined fastqs over to the bucket
+    try:
+        call(["gsutil", "-q", "-m", "mv", os.path.join('fastqs',"Undetermined_*"), os.path.join(output_directory,flowcell+"_Undetermined","")])
+        
+        # write the undetermined fastqs path for cromwell
+        file = open("undetermined.txt","w") 
+        file.write(os.path.join(output_directory,flowcell+"_Undetermined",""))
+        file.close()
+    except:
+        print("Unable to move Undetermined to Bucket")
+    
+    
+def orchestra_parse_csv(masterCsv):
+    """Parse the initial csv input to orchestra.
+        Arguments:
+        masterCsv- Csv File mapping samples, lanes and indices to bcl
+        Outputs-
+        samples_file- list of samples for scattering
+        bcl_file- list of bcl gsurls for scattering
+    """
+    # Read the masterCsv
+    df = pd.read_csv(masterCsv,header=0)
+    # Get unique gsurls of bcls
+    bcl_paths = set(df['Flowcell'].tolist())
+    # Write the bcls for Cromwell
+    bcl_file = open('bcls.txt', 'w+')
+    for item in bcl_paths:
+      bcl_file.write("%s\n" % item)
+    bcl_file.close()
+    # Get unique sample names
+    sampleIds = set(df['Sample'].tolist())
+    # Write the sample names for Cromwell
+    samples_file = open('samples.txt', 'w+')
+    for item in sampleIds:
+      samples_file.write("%s\n" % item)
+    samples_file.close()
+
+
+def orchestra_analysis_csv(masterCsv, h5s):
+    """Map the initial csv input to CellRanger Count h5 file outputs for analysis WDL.
+        Arguments:
+        masterCsv- Csv File mapping samples, lanes and indices to bcl
+        h5s- array of cromwell localized file strings of h5 files from CellRanger Count
+        Outputs-
+        analysis_csv- list of samples for scattering
+    """
+    # Read master Csv
+    df = pd.read_csv(masterCsv,header=0)
+    # We don't need Flowcell, Lane or Index for Bo's Wdl
+    df = df.drop(columns = ["Flowcell", "Lane", "Index"])
+    # Sort by Sample Name
+    df = df.sort_values(by=["Sample"]).drop_duplicates(subset=["Sample"])
+    sorted_h5s = []
+    # TODO this is more robust but still not good enough
```

Comment:
Fixed. Basically, calling "sample in" might not work if your sample names are something like 'test_sample' and 'test_sample2'. This can always be fixed manually and re run with call caching.

Reply:
Very helpful.  Could you add that to your comment in the code?

Reply:
Yep, added to code